### PR TITLE
[15.0][FIX] github_connector*: Misc changes

### PR DESCRIPTION
**Pull request by victoralmau**
Misc changes:
- [x] `github_connector`: Code refactor.
- [x] `github_connector_odoo`: Code refactor.
- [x] `github_connector_odoo`: Set the correct access rules for `odoo.module.version.rule.info`

Locked by:
- [x] https://github.com/OCA/interface-github/pull/105

Please @pedrobaeza can you review it?

@Tecnativa

### CODE REVIEW

File: github_connector/models/github_repository_branch.py [hunk 1]
```diff
@@ -227,6 +227,13 @@ def _get_analyzable_files(self, existing_folder):
         return res
 
     def set_analysis_rule_info(self):
+        """Remove all analysis info previously + rule process.
```
Comment (pedrobaeza) [inline github_connector/models/github_repository_branch.py]: This is not a docstring, but a comment about the following line. Dosctrings is for documenting what the method does, which is not the case.

File: github_connector/models/github_repository_branch.py [hunk 2]
```diff
@@ -227,6 +227,13 @@ def _get_analyzable_files(self, existing_folder):
...
+        We need to delete all linked records in case any linked rule has been deleted."""
+        self.env["github.repository.branch.rule.info"].search(
+            [
+                ("repository_branch_id", "=", self.id),
+            ]
+        ).sudo().unlink()
```
Comment (pedrobaeza) [inline github_connector/models/github_repository_branch.py]: Simpler:

```suggestion
        self.analysis_rule_info_ids.sudo().unlink()
```

**Review by pedrobaeza — COMMENTED** [review]
I'm wondering if recreating everything each time is the best strategy.

**Comment by victoralmau** [comment]
Ok, I understand (although this is not related to these changes).

What approach do you propose? While perhaps not the best strategy, I think that if you click on the 'Analyze Source Code' button you should analyze it again in case something has changed (or with cron).

**Comment by pedrobaeza** [comment]
It's a bit related, as the patch should be to remove the lines of unexisting rules, but maybe it's better to put an ondelete="cascade" to let the system to do it.

And about the strategy, it should be:

- Create a new info line if not exists for the rule.
- Update the existing one if exists.

**Comment by victoralmau** [comment]
Code refactoring in both modules according to your comments.

**Review by pedrobaeza — APPROVED** [review]
Let's fix the analysis:

/ocabot merge minor

**Pull request merged**